com.unity.sg2/Editor/ContextLayeredDataStorage/ContextLayeredDataStorage.cs

```diff
@@ -540,11 +546,42 @@ public virtual (string layer, string metadata) CopyElementCollection(IEnumerable
             return (EditorJsonUtility.ToJson(serializedLayer, true), EditorJsonUtility.ToJson(meta, true));
         }
 
-        public virtual void PasteElementCollection(string serializedLayer, string serializedMetadata)
+        public virtual IEnumerable<DataReader> PasteElementCollection(string serializedLayer, string serializedMetadata, string layerToPasteTo, out Dictionary<string, string> remappings)
         {
             //WIP, need to work through the logic of pasting into an existing layer
             var pasteLayer = new SerializedLayerData();
             EditorJsonUtility.FromJsonOverwrite(serializedLayer, pasteLayer);
+
+            var meta = new MetadataCollection();
+            EditorJsonUtility.FromJsonOverwrite(serializedMetadata, meta);
+
+            var root = m_layerList.GetLayerRoot(layerToPasteTo);
+            remappings = new Dictionary<string, string>();
+            List<DataReader> addedElements = new List<DataReader>();
+            foreach (var elemData in pasteLayer.layerData)
+            {
+                var elem = DeserializeElement(elemData);
+                MetadataBlock block = null;
+                meta.TryGetValue(elem.ID.FullPath, out block);
```

Comment:
After this TryGetValue we don't actually use block until line 578.  Is it possible to move these two lines down closer to the usage site, or does the following code cause changes to the value of elem.ID.FullPath?

Reply:
The following code will change the elem.ID if a collision is found, so I do it there to grab it on the unchanged ID. I could cache off the unchanged ID and use it later if you prefer, though

Reply:
Nah, I think it's reasonable as is given that context.  I had a suspicion that was the case, but good to get the confirmation.